code/datums/quirks/positive_quirks.dm

```diff
@@ -362,3 +362,221 @@
+			you are awarded with a 25% hazard pay bonus due to your [on_a_planet ?  "station" : "occupational"] assignment."))
+
+	// Supply them with some patches to help out on their new assignment
+	var/obj/item/storage/pill_bottle/ondansetron/disgust_killers = new()
+	disgust_killers.desc += " Best to take one when travelling to a planet's surface."
+	quirk_holder.equip_to_slot_if_possible(disgust_killers, ITEM_SLOT_BACKPACK, qdel_on_fail = TRUE, initial = TRUE, indirect_action = TRUE)
+
+/datum/quirk/spacer_born/remove()
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED)
+
+	if(QDELING(quirk_holder))
+		return
+
+	quirk_holder.inertia_move_delay /= 0.8
+	quirk_holder.clear_mood_event("spacer")
+	quirk_holder.remove_movespeed_modifier(/datum/movespeed_modifier/spacer)
+	quirk_holder.remove_status_effect(/datum/status_effect/spacer)
+
+	var/mob/living/carbon/human/human_quirker = quirk_holder
+	human_quirker.set_mob_height(HUMAN_HEIGHT_MEDIUM)
+	human_quirker.physiology.pressure_mod /= 0.8
+	human_quirker.physiology.cold_mod /= 0.8
+
+/datum/quirk/spacer_born/proc/start_tracking(mob/living/tracking)
+	var/new_z = tracking.loc?.z
+	// Some maps may decide to spawn people on mining Z, so just cast a wide net.
+	if(!is_station_level(new_z) && !is_mining_level(new_z) && !is_centcom_level(new_z))
+		return FALSE
+
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED) // gets rid of arrived_at_station registration if present
+	RegisterSignal(quirk_holder, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(spacer_moved)) // replaces it with spacer_moved
+
+	// Yes, it's assumed for planetary maps that you start at gravity sickness.
+	check_z(quirk_holder, skip_timers = TRUE)
+	return TRUE
+
+/// Check on Z level change if we should start tracking the mob, esssentially to catch latejoiners or other oddities
+/datum/quirk/spacer_born/proc/arrived_at_station(mob/living/source, turf/old_turf, turf/new_turf, same_z_layer)
+	SIGNAL_HANDLER
+
+	start_tracking(source)
+
+/// Check on Z change whether we should start or stop timers
+/datum/quirk/spacer_born/proc/spacer_moved(mob/living/source, turf/old_turf, turf/new_turf, same_z_layer)
+	SIGNAL_HANDLER
+
+	check_z(source)
+	// Using Z moved because we don't urgently need to check on every single turf movement for planetary status.
+	// If you've arrived at a "planet", the entire Z is gonna be a "planet".
+	// It won't really make sense to walk 3 feet and then suddenly gain / lose gravity sickness.
+	// If I'm proven wrong, swap this to use Moved.
+
+/**
+ * Used to check if we should start or stop timers based on the quirk holder's location.
+ *
+ * * afflicted - the mob arriving / same as quirk holder
+ * * skip_timers - if TRUE, this is being done instantly / should not have feedback (such as in init)
+ */
+/datum/quirk/spacer_born/proc/check_z(mob/living/source, skip_timers = FALSE)
+	var/turf/open/new_turf = get_turf(source)
+
+	// What is a planet?
```

Comment:
if you're going to have sixteen lines of documentation to help narrow it down for a generic(ish) edge case, why not  make this a more generic global proc that can be used elsewhere in the code?